docs/design/operator.md

```diff
@@ -0,0 +1,200 @@
+# Gluster Operator 1.0 Design
+
+## Introduction
+
+This document specifies the overall design for a Gluster operator. An operator
+is a Kubernetes [controller](https://github.com/kubernetes/sample-controller)
+that is focused on managing a specific application with the goal of automating
```

Comment:
I dont think we have to specify the `sample controller` here, the reason being no bits about implementation is covered here in the PR. We have to discuss further on the implementation side and come up best suited controller model, there are different framework available to implement an operator ( meta controller , operator sdk, ..etc),  we have to discuss further on it and come up with an implementation doc or PR later.

Reply:
Find me a better link to showcase what a controller is and I'll happily change it. :)

Reply:
I've asked around about the operator SDK and it very much doesn't even exist yet. While it sounds like it might be useful, I don't feel like waiting around for it.

Reading on metacontroller, it looks like a utility API for writing controllers. It doesn't sound like it has anything particularly novel, and in some aspects looks more constricting than just writing the controller framework by hand. Given that I've already written a controller base, I'm not interested in exploring metacontroller at this time.

Reply:
> I've asked around about the operator SDK and it very much doesn't even exist yet. While it sounds like it might be useful, I don't feel like waiting around for it.

The SDK is here: https://github.com/operator-framework/operator-sdk
While it is still very actively changing, it is certainly worth looking at as a template. It is also probably going to be useful to be compatible with the lifecycle manager: https://github.com/operator-framework/operator-lifecycle-manager

Reply:
Okay, more accurately: I was told not to use it yet. :) It effectively does not exist for me until it at least hits alpha.